upstream_utils/README.md

```diff
@@ -20,72 +20,42 @@ versions. Each library has its own patch directory (e.g., `lib_patches`).
 The example below will update a hypothetical library called `lib` to the tag
 `2.0`.
 
-Start in the `upstream_utils` folder. Restore the original repo.
+Start in the `upstream_utils` folder. Make sure a clone of the upstream repo exists.
 ```bash
-./update_<lib>.py
+./<lib>.py clone
 ```
 
-Navigate to the repo.
+Rebase the clone of the upstream repo.
 ```bash
-cd /tmp/lib
+./<lib>.py rebase 2.0
 ```
 
-Fetch the desired version using one of the following methods.
+Update the `upstream_utils` patch files and the tag in the script.
 ```bash
-# Fetch a full branch or tag
-git fetch origin 2.0
-
-# Fetch just a tag (useful for expensive-to-clone repos)
-git fetch --depth 1 origin tag 2.0
-```
-
-Rebase any patches onto the new version. If the old version and new version are
-on the same branch, run the following.
-```bash
-git rebase 2.0
-```
-
-If the old version and new version are on different branches (e.g.,
-llvm-project), use interactive rebase instead and remove commits that are common
-between the two branches from the list of commits to rebase. In other words,
-only commits representing downstream patches should be listed.
-```bash
-git rebase -i 2.0
-```
-
-Generate patch files for the new version.
-```bash
-git format-patch 2.0..HEAD --zero-commit --abbrev=40 --no-signature
-```
-
-Move the patch files to `upstream_utils`.
-```
-mv *.patch allwpilib/upstream_utils/lib_patches
-```
-
-Navigate back to `upstream_utils`.
-```bash
-cd allwpilib/upstream_utils
+./<lib>.py format-patch 2.0
```

Comment:
The tag argument no longer exists. The comment seems correct though.

Reply:
Do we want to completely remove it or still allow explicitly specifying it?

Reply:
I don't think there's a use case for that. There's no point in generating patches on top of a branch you aren't rebased upon.